pkg/steps/new_e2e.go

```diff
@@ -181,6 +199,50 @@ func imageForStep(image string, params api.Parameters) (string, error) {
 	return image, nil
 }
 
+func addSecretMount(pod *coreapi.Pod) {
+	path := "/tmp/installer"
+	pod.Spec.Volumes = append(pod.Spec.Volumes, coreapi.Volume{
+		Name: "installer",
+		VolumeSource: coreapi.VolumeSource{
+			Secret: &coreapi.SecretVolumeSource{SecretName: "installer"},
+		},
+	})
+	volumeMounts := &pod.Spec.Containers[0].VolumeMounts
+	*volumeMounts = append(*volumeMounts, coreapi.VolumeMount{
+		Name:      "installer",
+		MountPath: path,
+	})
+}
+
+func createSecret(secretClient coreclientset.SecretsGetter, namespace, artifactDir string) (string, error) {
+	if artifactDir == "" {
+		return "", nil
+	}
+	dir := filepath.Join(artifactDir, "output")
+	files, err := ioutil.ReadDir(dir)
+	if os.IsNotExist(err) {
+		return "", nil
+	} else if err != nil {
+		return "", err
+	}
+	secret := coreapi.Secret{
+		ObjectMeta: meta.ObjectMeta{Name: "installer"},
+		Data:       make(map[string][]byte, len(files)),
+	}
+	for _, f := range files {
```

Comment:
This seems fragile -- if the secret is write-able (or if we use a configmap) the pre-steps can just self-manage this